pahfit/base.py

```diff
@@ -512,14 +570,139 @@ def save(self, obs_fit, filename, outform):
+        cftable = Table(
+            names=("Name", "x_0_min", "x_0_max", "strength", "strength_unc", "eqw"),
+            dtype=("U25", "float64", "float64", "float64", "float64", "float64"))
+
+        # Get indices of dust features.
+        df_ind = np.concatenate(np.argwhere(ftable["Form"] == "Drude1D"))
+
+        # Get table subset containing only the dust features.
+        dfs = ftable[df_ind[0]:df_ind[-1] + 1]
+
+        # Iterate keys and calculate combined strengths.
+        for feat in cfdic.keys():
+            mask = np.logical_and(dfs['x_0'] >= cfdic[feat]['range'][0], dfs['x_0'] <= cfdic[feat]['range'][-1])
+            cftable.add_row(
+                [feat,
+                 cfdic[feat]['range'][0],
+                 cfdic[feat]['range'][-1],
+                 np.sum(dfs[mask]['strength']),
+                 None,
+                 np.sum(dfs[mask]['eqw'])
+                 ])
+        return cftable
+
+    @staticmethod
+    def eqws(comp_type, x_0, amp, fwhm_stddev, obs_fit):
+        """
+        Calculate the emission features equivalent width in microns.
+
+        Parameters
+        ----------
+        comp_type : string
+            type of emission component (Drude1D/Gaussian)
+        x_0 : float
+            central wavelength of the feature.
+        amp : float
+            central intensity of the feature.
+        fwhm_stddev : float
+            fwhm or stddev of the feature depending on comp_type.
+
+        Returns
+        -------
+        eqw : float
+            the equivalent width of the feature
+        """
+        # Check if the emission component is Gaussian and calculate fwhm.
+        if comp_type == 'Gaussian1D':
+            fwhm = 2 * fwhm_stddev * np.sqrt(2 * np.log(2))
+        else:
+            fwhm = fwhm_stddev
+
+        # Get range and wavelength region for integration.
+        low = x_0 - (fwhm * x_0 * 6)
+        lmin = low if low > 0 else 0.
+        lmax = x_0 + (fwhm * x_0 * 6)
+        lam = np.arange(100) / 99 * (lmax - lmin) + lmin
+
+        # Calculate the continuum and feature components in the integration range.
+        cont_components = []
+        for cmodel in obs_fit:
+            if isinstance(cmodel, BlackBody1D):
+                cont_components.append(cmodel)
+        cont_model = cont_components[0]
+        for cmodel in cont_components[1:]:
+            cont_model += cmodel
+        continuum = np.nan_to_num(cont_model(lam))
+
+        if comp_type == 'Drude1D':
+            drude = Drude1D(amplitude=amp,
+                            x_0=x_0,
+                            fwhm=fwhm)
+            lnu = drude(lam)
+
+        elif comp_type == 'Gaussian1D':
+            gauss = Gaussian1D(amplitude=amp,
+                               mean=x_0,
+                               stddev=fwhm_stddev)
+            lnu = gauss(lam)
+
+        # Set default broad limit value.
+        bl = 0.05
+
+        # Calculate EQW.
+        if fwhm > bl:
```

Comment:
In IDL PAHFIT, the condition is fractional FWHM > bl .... should it be the same here?

Reply:
It depends if `st.fwhm` in line 598 of pahfit.pro is the standard fwhm or the fractional. If it is the fractional then yes I should change this here.